src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,156 @@
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String INVALID_PASSWORD = "pass";
+    private static final int MAX_PASSWORD_LENGTH = 25;
+    private static final String TOO_LONG_PASSWORD = "a".repeat(MAX_PASSWORD_LENGTH + 1);
+    private static final int VALID_AGE = 21;
+    private static final int INVALID_AGE = 17;
+    private static final int GREATER_THAN_MAX_AGE = 101;
+    private static final String EMPTY_STRING = "";
+    private static RegistrationServiceImpl registrationServiceImpl;
+    private static StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationServiceImpl = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void afterEach() {
+        Storage.people.clear();
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = createUser(null, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_emptyLogin_notOk() {
+        User user = createUser(EMPTY_STRING, EMPTY_STRING, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = createUser(VALID_LOGIN, null, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_emptyPassword_notOk() {
+        User user = createUser(VALID_LOGIN, EMPTY_STRING, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_nullAge_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, null);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_tooLongLogin_notOk() {
+        User user = createUser(TOO_LONG_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_existingLogin_notOk() {
+        User user = createUser(EXISTING_LOGIN, VALID_PASSWORD, VALID_AGE);
+        storageDao.add(user);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_invalidLogin_notOk() {
+        User user = createUser(INVALID_LOGIN, INVALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    public void register_validLogin_ok() throws RegistrationException {
```

Comment:
```suggestion
    public void register_validLogin_ok() {
```
remove from method definition everywhere